--- a/ros/core/rosbuild/private.cmake~	2011-02-25 18:59:33.000000000 -0500
+++ b/ros/core/rosbuild/private.cmake	2011-04-01 13:49:24.428414107 -0400
@@ -188,7 +188,7 @@
 
   # Check that the file exists, #1621
   set(_file_name _file_name-NOTFOUND)
-  find_file(_file_name ${file} ${PROJECT_SOURCE_DIR} /)
+  find_file(_file_name ${file} ${PROJECT_SOURCE_DIR} / NO_CMAKE_FIND_ROOT_PATH)
   if(NOT _file_name)
     message(FATAL_ERROR "Can't find rostest file \"${file}\"")
   endif(NOT _file_name)
@@ -228,7 +228,7 @@
 
   # Check that the file exists, #1621
   set(_file_name _file_name-NOTFOUND)
-  find_file(_file_name ${file} ${PROJECT_SOURCE_DIR} /)
+  find_file(_file_name ${file} ${PROJECT_SOURCE_DIR} / NO_CMAKE_FIND_ROOT_PATH)
   if(NOT _file_name)
     message(FATAL_ERROR "Can't find pyunit file \"${file}\"")
   endif(NOT _file_name)
@@ -265,7 +265,7 @@
 macro(_rosbuild_add_roslaunch_check file)
   # Check that the file exists, #1621
   set(_file_name _file_name-NOTFOUND)
-  find_file(_file_name ${file} ${CMAKE_CURRENT_SOURCE_DIR} /)
+  find_file(_file_name ${file} ${CMAKE_CURRENT_SOURCE_DIR} / NO_CMAKE_FIND_ROOT_PATH)
   if(NOT _file_name)
     message(FATAL_ERROR "Can't find roslaunch file or directory \"${file}\"")
   endif(NOT _file_name)
@@ -345,7 +345,7 @@
   set(_srclist ${ARGN})
   foreach(_src ${_srclist})
     set(_file_name _file_name-NOTFOUND)
-    find_file(_file_name ${_src} ${CMAKE_CURRENT_SOURCE_DIR} /)
+    find_file(_file_name ${_src} ${CMAKE_CURRENT_SOURCE_DIR} / NO_CMAKE_FIND_ROOT_PATH)
     if(NOT _file_name)
       message("[rosbuild] Couldn't find source file ${_src}; assuming that it is in ${CMAKE_CURRENT_SOURCE_DIR} and will be generated later")
       set(_file_name ${CMAKE_CURRENT_SOURCE_DIR}/${_src})
