--- ros/stacks/ros_comm/clients/cpp/roscpp_serialization/include/ros/serialization.h	(revision 13188)
+++ ros/stacks/ros_comm/clients/cpp/roscpp_serialization/include/ros/serialization.h	(working copy)
@@ -175,12 +175,12 @@
   { \
     template<typename Stream> inline static void write(Stream& stream, const Type v) \
     { \
-      *reinterpret_cast<Type*>(stream.advance(sizeof(v))) = v; \
+      memcpy(stream.advance(sizeof(v)), &v, sizeof(v)); \
     } \
     \
     template<typename Stream> inline static void read(Stream& stream, Type& v) \
     { \
-      v = *reinterpret_cast<Type*>(stream.advance(sizeof(v))); \
+      memcpy(&v, stream.advance(sizeof(v)), sizeof(v)); \
     } \
     \
     inline static uint32_t serializedLength(const Type t) \
